conda_declarative/apply.py

```diff
@@ -175,21 +175,12 @@ def apply(
     if prefix is None:
         prefix = context.target_prefix
 
-    env = from_env_file(str(prefix))
-
-    if env is not None:
-        requested_packages = env.requested_packages
-        if env.config is not None:
-            channels = env.config.channels
-    else:
-        channels = []
-        requested_packages = []
-
+    model = from_env_file(str(prefix))
     records = solve(
```

Comment:
I think this solve needs to also be pruning packages at some point. Currently if you add a package using `conda edit`, conda will do the right thing and install the package. If you remove the same package using `conda edit`, the package is not removed from the environment. 

I *think* using the prune option here will not remove all the packages required for conda to work. So I think this is safe. Would need to double check that though.

Reply:
Agreed. I've not really touched the actual installation/solving machinery yet, so I'll leave this for a future date - this PR is big enough 😅